tests/utils/test_rest_utils.py

```diff
@@ -397,3 +397,43 @@ def test_can_parse_as_json_object():
     assert not _can_parse_as_json_object("[0, 1, 2]")
     assert not _can_parse_as_json_object('"abc"')
     assert not _can_parse_as_json_object("123")
+
+
+def test_http_request_customize_config():
+    with mock.patch(
+        "mlflow.utils.rest_utils._get_http_response_with_retries"
+    ) as mock_get_http_response_with_retries:
+        host_only = MlflowHostCreds("http://my-host")
+        with mock.patch.dict(os.environ, {}, clear=True):
```

Comment:
You can use https://docs.pytest.org/en/7.1.x/how-to/monkeypatch.html

Reply:
I prefer to use patch dict here so that it can patch multiple key values easily